### ISSUE TITLE
Bump SDK to 3.0.4

### ISSUE THREAD

**Issue by marcelstoer** [nodemcu/nodemcu-firmware#3143]
https://github.com/espressif/ESP8266_NONOS_SDK/releases/tag/v3.0.4 was released today.

**Comment by TerryE** [comment]
OK, I will do this.

**Comment by nwf** [comment]
Can I request that this be deferred until after #3066?  I feel like we're never going to cut to master because things could always be a little bit better than they are.

**Comment by TerryE** [comment]
I had though of rolling this into #3193, but on reflection, I would prefer to do this on a standalone PR.

**Comment by marcelstoer** [comment]
Just realized this is a dup of #3008 -> closing
